src/Utilities/constraints.jl

```diff
@@ -10,20 +10,26 @@ the set in `model`. If `allow_modify_function` is `true` then the function
 """
 function add_scalar_constraint end
 
-function add_scalar_constraint(model::MOI.ModelLike, func::MOI.SingleVariable,
-                               set::MOI.AbstractScalarSet;
-                               allow_modify_function::Bool=false)
-    return MOI.add_constraint(model, func, set)
-end
 function add_scalar_constraint(model::MOI.ModelLike,
-                               func::Union{MOI.ScalarAffineFunction{T},
-                                           MOI.ScalarQuadraticFunction{T}},
+                               func::MOI.AbstractScalarFunction,
                                set::MOI.AbstractScalarSet;
                                allow_modify_function::Bool=false) where T
+    return MOI.add_constraint(
+        model, normalize_constant(
+            func, set; allow_modify_function=allow_modify_function)...)
+end
+
+function normalize_constant(func::MOI.AbstractFunction, set::MOI.AbstractSet;
```

Comment:
This name doesn't make it clear what's being normalized to what. Maybe `move_function_constant_to_set`?

Reply:
It's only moved in the scalar case. It's also consistent with: https://github.com/JuliaOpt/JuMP.jl/pull/1960
Maybe a docstring would make it clear ?